### PR TITLE
Migrate to .NET Core

### LINKED ISSUE #181: Build on CoreCLR / .Net Core
Hi,

First things first: Thanks for the great project!

Are there any plans to make this project CoreCLR compatible? There are just a couple of things that prevent it from building against CoreCLR. I did a quick hack here:

https://github.com/gabriel-samfira/dotliquid/commit/db43fd6dc3dca4811bdc365d45172399f4c6f67e

to build against the Windows Nano server TP4 build of CoreCLR.

Thanks!

### PR DESCRIPTION
**Pull request by tbolon**
Resolves #181

- [x] Add CORE specific compilation symbol to hide full Framework code
- [x] Convert csproj to xproj/project.json
- [x] Add both net45 and netstandard1.3 targets for DotLiquid
- [x] Migrate NUnit tests to latest .net core version
- [x] Migrate NUnit to dotnet runner
- [x] Remove obsolete VS solutions and projects (2008/2010)
- [x] Convert Website to ASP.NET Core
- [x] Remove nuspec / update project.json
- [x] Update appveyor build
- [x] Update travis build
- [x] fix nuget package numerotation on appveyor

Project now requires VS 2015 (update 3 ?)

### CODE REVIEW

**Comment by haf** [comment]
You can look at how Suave is doing it in its rakefile on Appveyor vs linux

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
+dotnet: 1.0.0-preview2-003121
+mono: none
 before_install:
   - sudo apt-get install nunit-console
 script:
- - xbuild /p:Configuration=Release src/DotLiquid.sln
- - nunit-console src/DotLiquid.Tests/bin/Release/DotLiquid.Tests.dll
+  - dotnet restore
+  - dotnet build **/project.json --configuration Release
+  - nunit-console src/DotLiquid.Tests/bin/Release/netcoreapp1.0/DotLiquid.Tests.dll src/DotLiquid.Tests/bin/Release/net451/win7-x64/DotLiquid.Tests.dll
```
Comment (laedit) [inline .travis.yml]: Why don't use dotnet test?

**Comment by laedit** [comment]
Artifacts are missing on AppVeyor.

**Comment by laedit** [comment]
About the nuget packages version: `dotnet pack`doesn't support a `version` argument and it seems that it will be [removed](https://github.com/dotnet/cli/issues/2676).
So I think that you can use `nuget pack` directly, which have a `version` argument.

**Comment by tbolon** [comment]
@laedit you are right, I have restored the nuspec file, and now using nuget pack.

I have also fixed the problem with the version number. Perhaps the nuget push will not work, but I am unable to test it as you have to be in master branch to use it.

**Comment by laedit** [comment]
Sounds all good! 😃 
Thanks!

**Comment by haf** [comment]
Let's get this merged then!

**Pull request merged**